### PR TITLE
Relax constraints on return statements in the script

### PR DESCRIPTION
**Pull request by apaszke**
Script functions can now have no return statements, empty
return statements, or return one or more values.

Additionally fix the lexer to always emit TK_NEWLINE before
TK_DEDENT, which simplifies the parser.

### CODE REVIEW

**Comment by apaszke** [comment]
Fixes #6064.

File: test/test_jit.py [hunk 1]
```diff
@@ -1802,6 +1801,25 @@ def func(a):
...
+            return
+
+        def one_return(a):
+            return a + 1.
+
+        def multiple_returns(a):
+            return a * 1., a * 2., a * 3.
+
+        a = torch.randn(1, dtype=torch.float)
+        # self.checkScript(no_return, [a], optimize=True)
```
Comment (zdevito) [inline test/test_jit.py]: commented out?

File: torch/csrc/jit/script/lexer.h [hunk 2]
```diff
@@ -373,48 +373,43 @@ struct Token {
 };
 
 struct Lookahead {
-  Lookahead(const Token& t) : t(t) {}
-  Token t;
+  std::vector<Token> tokens;
```
Comment (zdevito) [inline torch/csrc/jit/script/lexer.h]: I think this struct is dead.

**Review by zdevito — APPROVED** [review]
Looks good. One minor comment. Thanks for fixing the lexer!

**Comment by ezyang** [comment]
@apaszke merge at your leisure

**Pull request merged**